airflow/providers/slack/operators/slack.py

```diff
@@ -203,13 +213,31 @@ def __init__(
         self.filename = filename
         self.filetype = filetype
         self.content = content
+        self.file_params = {}
         super().__init__(method=self.method, **kwargs)
 
     def construct_api_call_params(self) -> Any:
-        self.api_params = {
-            'channels': self.channel,
-            'content': self.content,
-            'filename': self.filename,
-            'filetype': self.filetype,
-            'initial_comment': self.initial_comment,
-        }
+        if self.content is not None:
+            self.api_params = {
+                'channels': self.channel,
+                'content': self.content,
+                'initial_comment': self.initial_comment,
+            }
+        elif self.filename is not None:
+            self.api_params = {
+                'channels': self.channel,
+                'filename': self.filename,
+                'filetype': self.filetype,
+                'initial_comment': self.initial_comment,
+            }
+            self.file_params = {'file': self.filename}
+
+    def execute(self, **kwargs):
+        """
+        The SlackAPIOperator calls will not fail even if the call is not unsuccessful.
+        It should not prevent a DAG from completing in success
+        """
+        if not self.api_params:
+            self.construct_api_call_params()
+        slack = SlackHook(token=self.token, slack_conn_id=self.slack_conn_id)
+        slack.call(self.method, data=self.api_params, files=self.file_params)
```

Comment:
Why do we need to pass `filename ` in both `files` & `data` ?
`api_params` has a `filename` key that contains `self.filename`
`file_params` has `file` key that contains `self.filename`